number_theory/numeric.h

```diff
@@ -55,7 +55,7 @@ std::pair<T, T> exgcd(T a, T b) {
 template <typename T,
           typename U,
           std::enable_if_t<std::numeric_limits<U>::is_integer, bool> = true>
-T pow(const T &base, U exponent) {
+T pow(T base, const U &exponent) {
```

Comment:
Why are we switching `base` to pass-by-value?

Reply:
Notice that base is only used once in this function. If the user pass a rvalue to pow, we can achieve zero-copy.